doc/developer_guild.md

```diff
@@ -0,0 +1,471 @@
+# Extend Functionalities of GoTorch
+
+At the very core of PyTorch is *libtorch*, which is a C++ library that contains
+three levels of abstraction:
+
+1. *Native function*s are low-level functions, most of which operate tensors.
+   They are the basic mathematical build blocks of deep neural networks.
+1. *Functional*s are high-level functions built on native functions, each
+   functional may call several native functions.
+1. *Module*s are c++ classes/structs/templates that encapsulate multiple native
+   functions or functionals.
+
+Modules and functionals form the Pytorch C++ frontend.  AI practitioners can
```

Comment:
According to https://pytorch.org/cppdocs/frontend.html, PyTorch C++ frontends includes

- torch::Tensor, Automatically differentiable, efficient CPU and GPU enabled tensors
- torch::nn, A collection of composable modules for neural network modeling
- torch::optim, Optimization algorithms like SGD, Adam or RMSprop to train your models
- torch::data, Datasets, data pipelines and multi-threaded, asynchronous data loader
- torch::serialize, A serialization API for storing and loading model checkpoints
- torch::python, Glue to bind your C++ models into Python
- torch::jit, Pure C++ access to the TorchScript JIT compiler

These include but not limited to modules or functionals.

I don't think it is necessary to mention the concept of PyTorch C++ frontend.

Reply:
Done.